utils/components.py

```diff
@@ -298,3 +301,302 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+            #####################################################################################
+            """)
+            sys.exit(1)
+        print(f"""
+        ########### INFO ##############
+        # Model Registry is available #
+        ###############################
+        """)
+    except Exception as e:
+        print(f"""
+        ############# ERROR ###############
+        # Model Registry is not available #
+        ###################################
+        """)
+        sys.exit(1)
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_training_operator():
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+    ) as namespace_path:
+        namespace = namespace_path.readline()
+    config.load_incluster_config()
+
+    with client.ApiClient() as api_client:
+        api_instance = client.CustomObjectsApi(api_client)
+        group = "kubeflow.org"
+        version = "v1"
+        plural = "pytorchjobs"
+
+        try:
+            api_response = api_instance.list_namespaced_custom_object(
+                group, version, namespace, plural
+            )
+            print("""
+            ######################### INFO ###########################
+            # Kubeflow Training Operator PyTorchJob CRD is available #
+            ##########################################################
+            """)
+        except ApiException as e:
+            print("""
+            #################################################### ERROR ######################################################################
+            # Kubeflow Training Operator PyTorchJob CRD is unavailable. Ensure your OpenShift AI installation has Training Operator enabled #
+            #################################################################################################################################
+            """)
+            sys.exit(1)
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_oci_model(output_oci_model_uri: str, output_oci_registry_secret: str):
+    import base64
+    import json
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+    ) as namespace_path:
+        namespace = namespace_path.readline()
+    config.load_incluster_config()
+
+    if output_oci_model_uri is None:
+        print(f"""
+        ############################## INFO ##################################
+        # Parameter output_oci_model_uri not provided. Skipping this step... #
+        ######################################################################
+        """)
+        sys.exit(0)
+    elif not output_oci_model_uri.startswith("oci://"):
```

Comment:
This is actually okay because `upload_model_op` adds the `oci://` prefix if it's missing. This is okay because the pipeline parameter includes `oci` in the name, so it implies that any user value for it means they want to push the model as a Modelcar.